sdk/src/beta9/runner/endpoint.py

```diff
@@ -209,6 +226,7 @@ def __init__(self, logger: logging.Logger, worker: UvicornWorker) -> None:
             self.app = FastAPI(lifespan=self.lifespan)
 
         self.app.add_middleware(TaskLifecycleMiddleware)
+        self.app.add_middleware(WebsocketTaskLifecycleMiddleware)
```

Comment:
I guess we need to add both - they don't conflict do they? Like we're not going to accidentally be using the websocket middleware for non-websockets. Also another concern is overhead of two layers of middleware, are they both executed (or attempted to be executed) everytime?

Reply:
both are being executed but it immediately skips the websocket one cause of this
```
if scope["type"] != "websocket":
            await self.app(scope, receive, send)
            return

```

The TaskMiddleware behind the scenes already do
```
if scope["type"] != "http":
            await self.app(scope, receive, send)
            return
```